test/e2e/e2e_test.go

```diff
@@ -51,4 +52,31 @@ func TestE2E(t *testing.T) {
 	cSuite.Setup(t)
 	t.Logf("Running %d E2E tests", len(tests.ConformanceTests))
 	cSuite.Run(t, tests.ConformanceTests)
+
+	// E2E tests for other GatewayClasses.
+	NewE2ETestForGatewayClass(t, "MergeGateways E2E", "merge-gateways", client, tests.MergeGatewaysTests)
```

Comment:
why did we need to create a new test suite ?

Reply:
nvm I see the issue

Reply:
we'll need to revisit this code, to add e2e for multiple GC, one thought is to update
https://github.com/kubernetes-sigs/gateway-api/blob/d58ecdeacc0f516869f354cdd15d13bd5b9bd8fe/conformance/utils/kubernetes/apply.go#L184
to skip `preparing` resources which have a `gateway-api/dont-edit` annotation similar to
https://github.com/kubernetes-sigs/gateway-api/blob/d58ecdeacc0f516869f354cdd15d13bd5b9bd8fe/conformance/utils/kubernetes/helpers.go#L47
&&
https://github.com/kubernetes-sigs/gateway-api/blob/d58ecdeacc0f516869f354cdd15d13bd5b9bd8fe/conformance/tests/gateway-static-addresses.yaml#L8

Reply:
that would be great to have this apply logic modified to allow such case